internal/api/schemas/schemas.go

```diff
@@ -178,15 +180,45 @@ func (s *sch) ValidateQuery(plugin common.Plugin, queryName string) error {
 	return s.validatePlugin(plugin, "query", queryName, s.queries)
 }
 
-func (s *sch) ValidateGlobalVariable(v modelV1.VariableSpec) error {
-	if v.Kind != variable.KindList {
-		return nil
+func (s *sch) ValidateVariableSpec(v modelV1.VariableSpec) error {
+	switch v.Kind {
+	case variable.KindDatasource:
+		return s.validateDatasourceVariable(v)
+	case variable.KindList:
+		return s.validateListVariable(v)
+	case variable.KindText:
+		return nil // no extra validation needed here
+	default:
+		return errors.New("unsupported variable kind")
+	}
+}
+
+func (s *sch) validateDatasourceVariable(v modelV1.VariableSpec) error {
+	datasourceVariableSpec, ok := v.Spec.(*variable.DatasourceSpec)
+	if !ok {
+		return errors.New("Error converting Variable to DatasourceVariable")
 	}
 
+	schemasPointer := atomic.LoadPointer((*unsafe.Pointer)(unsafe.Pointer(&s.dts.schemas)))
```

Comment:
why don't you use `s.dts.schemas.Load` instead ?

Reply:
Ah you're right, I forgot the schemas map is already an atomic.Pointer.. I'll change that.